sycl/include/sycl/stl_wrappers/complex

```diff
@@ -0,0 +1,91 @@
+#pragma once
+
+#include_next <complex>
```

Comment:
Long time since I have not seen this in use.
Perhaps a good opportunity to add some comments before explaining the basic design ideas?

Reply:
Done, I think.

Reply:
Is `#include_next` supported by MSVC?  A cursory google search says it is not.